finder/tagged.go

```diff
@@ -226,14 +262,54 @@ func ParseTaggedConditions(conditions []string) ([]TaggedTerm, error) {
 		default:
 			return nil, fmt.Errorf("wrong seriesByTag expr: %#v", s)
 		}
+		if len(taggedCosts) > 0 {
+			if costs, ok := taggedCosts[terms[i].Key]; ok {
+				setCost(&terms[i], costs)
+			}
+		}
 	}
 
-	sort.Sort(TaggedTermList(terms))
+	if len(taggedCosts) == 0 {
+		sort.Sort(TaggedTermList(terms))
+	} else {
+		// compare with taggs costs
+		sort.Slice(terms, func(i, j int) bool {
+			eq, comparable := lessCosts(terms, i, j)
+			if comparable {
+				return eq
+			}
+
+			if terms[i].Op < terms[j].Op {
```

Comment:
I would like to see more comments about what this code should do. As I'm a bit lost trying to understand why comparing one TaggedTermOp with another is a good sorting condition (I'm actually wonder how good that clause is and how different Term would affect performance, as I'm not sure if Match vs Not Match matters, as well as Eq and Not Eq likely have same cost for us)